### PR TITLE
Remove Jekyll Site

### PR DESCRIPTION
**Pull request by AndyClifford**
Removing the Jekyll site as we are no longer serving this content. Also updated the integration tests to extract h2's from our pages instead of anchor tags. This is more representative of the content. Changing an H2 (e.g. /api/payments#my-h2) will break the integration test.

**Ticket Link:** https://www.notion.so/centrapay/Release-new-API-layout-in-production-9e3974fee0d140eda47b6996e04f829c?pvs=4

**Test Plan:**
- [ ] CI Pipeline passes
- [ ] <baseUrl>/api/introduction.html renders the astro 404 page
- [ ] Run Deploy Dev pipeline and repeat steps 1 and 2.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -9,27 +9,16 @@ jobs:
...
         with:
           node-version: 18
       - run: yarn install --frozen-lock-file
       - run: yarn lint
       - run: yarn test
-      - run: ./build.sh --mode production
+      - run: ./build.sh
       - uses: JamesIves/github-pages-deploy-action@3.7.1
         with:
           GITHUB_TOKEN: ${{ secrets.JEKYLL_PAT }}
```
Comment (tom-blake) [inline .github/workflows/publish.yml:22]: @AndyClifford Suggestion: Remove all references to Jekyll

File: build.sh [hunk 2]
```diff
@@ -14,19 +14,10 @@ while [ $# -gt 0 ]; do
...
 
-cd legacy
-
-bundle install
-bundle exec jekyll build
-
-cd ../
-
 if [ "$mode" == "development" ]; then
   yarn build --mode "$mode" --site "http://centrapay-docs.dev.s3-website-ap-southeast-1.amazonaws.com"
```
Comment (tom-blake) [inline build.sh:18]: @AndyClifford Suggestion: Set `--mode` and `--site` in the Github actions and delete this script

File: .github/workflows/publish.yml (hunk 1 above)
Comment (AndyClifford) [reply .github/workflows/publish.yml:22]: Can you please duplicate the value of `JEKYLL_PAT` to `GITHUB_TOKEN`

File: build.sh (hunk 2 above)
Comment (AndyClifford) [reply build.sh:18]: And copy the CNAME in Github workflow too?

**Review by alexBinns — APPROVED** [review]

**Pull request merged**